src/WebJobs.Extensions.DocumentDB/Config/DocumentDBConfiguration.cs

```diff
@@ -13,14 +19,15 @@ namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
     public class DocumentDBConfiguration : IExtensionConfigProvider
     {
         internal const string AzureWebJobsDocumentDBConnectionStringName = "AzureWebJobsDocumentDBConnectionString";
+        internal readonly ConcurrentDictionary<string, IDocumentDBService> ClientCache = new ConcurrentDictionary<string, IDocumentDBService>();
 
         /// <summary>
         /// Constructs a new instance.
         /// </summary>
         public DocumentDBConfiguration()
         {
-            this.ConnectionString = GetSettingFromConfigOrEnvironment(AzureWebJobsDocumentDBConnectionStringName);
-            this.DocumentDBServiceFactory = new DefaultDocumentDBServiceFactory();
+            ConnectionString = GetSettingFromConfigOrEnvironment(AzureWebJobsDocumentDBConnectionStringName);
+            DocumentDBServiceFactory = new DefaultDocumentDBServiceFactory();
```

Comment:
We talked about this elsewhere -- but I wanted to call out that (I think) this is the only places that we still check Environment vars and Configuration directly anymore. It'd be nice to send these calls through a NameResolver (or similar).